### PR TITLE
Review logging

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Ensure enough logging to producer](https://tools.hmcts.net/jira/browse/RPE-311)

### Change description ###

There are continuous info logs around each task to assure app is running correctly. Minor minimisation applied by moving ftp availability check prior actual task run.
Since almost all service factors are covered by tasks, the other places are fairly covered by letter creations and built-in tracking: web requests, dependency hits defined in app insights' agent config.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java [hunk 1]
```diff
@@ -43,27 +43,28 @@ public MarkLettersPostedTask(LetterRepository repo,
     }
 
     public void run(LocalTime now) {
+        if (!ftpAvailabilityChecker.isFtpAvailable(now)) {
+            logger.info("Not processing letter postage report due to FTP downtime window");
+            return;
+        }
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java]: add `else` for clarity

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java [hunk 2]
```diff
@@ -42,13 +42,13 @@ public UploadLettersTask(
     }
 
     public void run() {
-        logger.info("Started letter upload job");
-
         if (!availabilityChecker.isFtpAvailable(now().toLocalTime())) {
-            logger.info("Not processing due to FTP downtime window");
+            logger.info("Not processing upload letter task due to FTP downtime window");
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java]: perhaps 
```
Not processing 'upload letter' task
```

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java [hunk 3]
```diff
@@ -43,27 +43,28 @@ public MarkLettersPostedTask(LetterRepository repo,
...
 
-                    if (parsedReport.allRowsParsed) {
-                        logger.info("Report {} successfully parsed, deleting", parsedReport.path);
-                        ftpClient.deleteReport(parsedReport.path);
-                    } else {
-                        logger.warn("Report {} contained invalid rows, file not removed.", parsedReport.path);
-                    }
-                });
-        } else {
-            logger.info("Not processing due to FTP downtime window");
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java]: logging remains the same, why this change??

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java]: As method evolves the nesting goes deeper and deeper making the indentation to 4th level. This is not needed when you can break the process at the very beginning then continue evolving/reading method normally without additional nesting wrap.
Makes sense to keep small if/else blocks. Here main part is actually responsible for something greater and making the block first level is natural

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java [hunk 4]
```diff
@@ -61,11 +61,11 @@ public void run(LocalTime now) {
                         logger.warn("Report {} contained invalid rows, file not removed.", parsedReport.path);
                     }
                 });
+
+            logger.info("Completed report processing job");
         } else {
-            logger.info("Not processing due to FTP downtime window");
+            logger.info("Not processing '{}' task due to FTP downtime window", Task.MarkLettersPosted);
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java]: Inconsistent with how this job/task is referred to in other logs in this class
(`report processing job` vs `'MarkLettersPosted' task`)
